FakeNFT/Scenes/Collection/Catalogue/CatalogueVC.swift

```diff
@@ -0,0 +1,138 @@
+import UIKit
+
+final class CatalogueVC: UIViewController {
+    
+    let collectionViewModels: CatalogueViewModel
+    let tableView = UITableView()
+    
+    private lazy var dataSource = makeDataSource()
+    
+    init(viewModel: CatalogueViewModel) {
+        self.collectionViewModels = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        setupTableView()
+        setupNavBar()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        collectionViewModels.updateListener = self
+        collectionViewModels.loadCollections()
+    }
+
+    private func setupTableView() {
+        tableView.dataSource = dataSource
+        tableView.delegate = self
+        tableView.register(CatalogueCell.self, forCellReuseIdentifier: CatalogueCell.identifier)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        
+        let guide = view.safeAreaLayoutGuide
+        let hInset: CGFloat = 16
+        
+        tableView.separatorStyle = .none
+        tableView.allowsSelection = false
+        tableView.layer.masksToBounds = false
+        tableView.scrollIndicatorInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: -hInset)
+        
+        view.addSubview(tableView)
+        
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: guide.topAnchor, constant: 21),
+            tableView.leadingAnchor.constraint(equalTo: guide.leadingAnchor, constant: hInset),
+            tableView.trailingAnchor.constraint(equalTo: guide.trailingAnchor, constant: -hInset),
+            tableView.bottomAnchor.constraint(equalTo: guide.bottomAnchor)
+        ])
+    }
+    
+    private func setupNavBar() {
+        let sortButton: UIBarButtonItem = {
+            let button = UIBarButtonItem()
+            button.tintColor = .asset(.main(.primary))
+            button.style = .plain
+            button.image = UIImage(named: "sortIcon")
+            button.target = self
+            button.action = #selector(didTapSortButton)
+            return button
+        }()
+        
+        navigationItem.rightBarButtonItem = sortButton
+    }
+    
+    private func makeDataSource() -> UITableViewDiffableDataSource<CatalogueVC.ViewModel, CatalogueVC.ViewModel> {
+        
+        return UITableViewDiffableDataSource<CatalogueVC.ViewModel, CatalogueVC.ViewModel>(tableView: tableView) {
+            (tableView: UITableView, indexPath: IndexPath, viewModel: CatalogueVC.ViewModel) -> UITableViewCell? in
```

Comment:
Дженерик тип компилятор выводит по декларации функции
```swift
        return UITableViewDiffableDataSource(tableView: tableView) {
            (tableView: UITableView, indexPath: IndexPath, viewModel: CatalogueVC.ViewModel) -> UITableViewCell? in
```

Параметры кложуры пишутся на той же строке с фигурной скобкой (swiftlint будет ругаться)
```swift
        return UITableViewDiffableDataSource(tableView: tableView) { (tableView: UITableView, indexPath: IndexPath, viewModel: CatalogueVC.ViewModel) -> UITableViewCell? in
```

Тип аргумента компилятор тоже выведет, полностью его писать не нужно

 ```swift
        return UITableViewDiffableDataSource(tableView: tableView) { tableView, indexPath, viewModel in
```

и вот мы уже почти влезаем в столбик 140 символов. Я бы тут остановилась, но можно кароче:

```suggestion
       return .init(tableView: tableView) { tableView, indexPath, viewModel in
```

Reply:
Впечатляет! Спасибо